clients/githubrepo/searchCommits.go

```diff
@@ -0,0 +1,66 @@
+package githubrepo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/go-github/v38/github"
+
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+type searchCommitsHandler struct {
+	ghClient *github.Client
+	ctx      context.Context
+	repourl  *repoURL
+}
+
+func (handler *searchCommitsHandler) init(ctx context.Context, repourl *repoURL) {
+	handler.ctx = ctx
+	handler.repourl = repourl
+}
+
+func (handler *searchCommitsHandler) search(request clients.SearchCommitsOptions) (clients.SearchCommitsResponse, error) {
+	if !strings.EqualFold(handler.repourl.commitSHA, clients.HeadSHA) {
+		return clients.SearchCommitsResponse{}, fmt.Errorf(
+			"%w: Search only supported for HEAD queries", clients.ErrUnsupportedFeature)
+	}
+	query, err := handler.buildQuery(request)
+	if err != nil {
+		return clients.SearchCommitsResponse{}, fmt.Errorf("handler.buildQuery: %w", err)
+	}
+
+	resp, _, err := handler.ghClient.Search.Commits(handler.ctx, query, &github.SearchOptions{ListOptions: github.ListOptions{PerPage: 100}})
+	if err != nil {
+		return clients.SearchCommitsResponse{}, fmt.Errorf("Search.Code: %w", err)
+	}
+
+	return searchCommitsResponseFrom(resp), nil
+}
+
+func (handler *searchCommitsHandler) buildQuery(request clients.SearchCommitsOptions) (string, error) {
```

Comment:
Would be great if you could add some tests for the `buildQuery` fn. See https://github.com/ossf/scorecard/blob/main/clients/githubrepo/search_test.go for example.

Reply:
Added tests for SearchCommitsHandler's `buildQuery` :+1: